include/openthread/platform/radio.h

```diff
@@ -946,6 +946,36 @@ otError otPlatRadioEnableCsl(otInstance *aInstance, uint32_t aCslPeriod, const o
  */
 void otPlatRadioUpdateCslSampleTime(otInstance *aInstance, uint32_t aCslSampleTime);
 
+/**
```

Comment:
Here we are adding a bunch of APIs for channel related info. To have a more flexible way and reduce the number of `otPlatRadio` APIs for this purpose I have a suggestion:
- We can define a `struct` to specify all the data/info for a channel.
- Add a single API  that sets the info for a channel.

This also allows us to extend this in future (add more fields if needed).

Example:
```
typedef struct otRadioChannelInfo
{
    bool   mSupported;
    bool   mPreferred;
    int8_t mMaxTxPower;
} otRadioChannelInfo;

void otPlatRadioSetChannelInfo(uint8_t aChannel, const otRadioChannelInfo *aChannelInfo);
```

Reply:
After discussing with @bukepo, we decided to remove the channel mask in platform api.